packages/block-editor/src/hooks/dimensions.js

```diff
@@ -56,19 +57,33 @@ export function DimensionsPanel( props ) {
 	] );
 
 	// Callback to reset all block support attributes controlled via this panel.
-	const resetAll = () => {
+	const resetAll = ( resetFilters = [] ) => {
 		const { style } = props.attributes;
 
-		props.setAttributes( {
-			style: cleanEmptyObject( {
+		let newAttributes = {
+			style: {
 				...style,
 				spacing: {
 					...style?.spacing,
 					margin: undefined,
 					padding: undefined,
 				},
-			} ),
+			},
+		};
+
+		// Injected controls can register additional functions to further adjust
+		// the attributes being reset.
+		resetFilters.forEach( ( resetFilter ) => {
+			newAttributes = resetFilter( newAttributes );
```

Comment:
I know this is probably all going to change, so feel free to ignore the following! 

If I'm reading this right, we're replacing all style attributes with the fill component's `resetAllFilter` return value.

Should we expect these components to be aware of and return a valid props object, including the `style` property?

I suppose there might be a situation in which a fill component might want to rewrite entire style object, but I can't think of any right now.

For the cover block, all we really want to do is clear the block's internal attributes, and let the dimension hook deal with cleaning up the rest, so maybe something like this would cover all bases:

```js
newAttributes = {
    ...newAttributes,
    ...resetFilter( newAttributes ),
};
```

That would avoid JS errors where the fill component doesn't return what we expect: the clear the empty object below expects a `style` property.

Reply:
Thanks @ramonjd I was looking a making a similar improvement just so far as updating this to only apply the filter's updates to the attributes object instead of replacing it. This would make the application of filters more robust and the filters themselves more similar to `setAttributes`.

As it stands currently, the [filters essentially do what you're snippet here does](https://github.com/WordPress/gutenberg/pull/34065/files#diff-c9b6422f332979151b7228eb2af812e84859338e6b4d5ef38e5341ef6e8f5a39R554-R559).

```js
resetAllFilter={ ( newAttributes ) => {
	return {
		...newAttributes,
		minHeight: undefined,
	};
} }
```

> Should we expect these components to be aware of and return a valid props object, including the style property?

What is passed to these filters could be different in different panels. So that could be a moving target.

> I suppose there might be a situation in which a fill component might want to rewrite entire style object, but I can't think of any right now.

For the dimensions support we are currently only manipulating the style object however for borders, colors, or typography supports, it will be other top-level attributes as well such as `borderColor`. The code here was a bit more generic than it needed to be for dimensions given I was also trialling it with the other block supports.

> That would avoid JS errors where the fill component doesn't return what we expect: the clear the empty object below expects a style property.

I believe `cleanEmptyObject` handles an `undefined` value as well. The use of `cleanEmptyObject` here was to avoid requiring every filter to also need to filter the object. With the proposed change I don't think this is a problem, it ensures that we unset the style attribute completely when it is empty and unneeded.

Reply:
My plan today is to remove the `DimensionControls` implementation in this PR, rebase it on top of https://github.com/WordPress/gutenberg/pull/34069, and then make the proposed tweak to the filters for the dimensions support. Following that I'll update https://github.com/WordPress/gutenberg/pull/34065 accordingly.